### PR TITLE
Use *io.File instead of io.Reader, so buffer isn't needed

### PR DESCRIPTION
**Pull request by ajdevries**

### CODE REVIEW

**Comment by bramvdbogaerde** [comment]
I understand what you want to do, but I think it can break some existing use cases of the library.

Lets say you have an HTTP service that provides a way for the users to upload files, in those cases the HTTP library probaly provides access to the file embedded in the request through a type that implements io.Reader, with the current signature the programmer can very easily upload that file via SSH by just passing that type, since it was an io.Reader. With your changes the programmer will be required to convert the type to a os.File, almost certainly by storing it on the local file system.

So in short, although this eliminates any visible buffering, I think it is a unsafe assumption that the programmer always passes a os.File. 

If you don't agree with my reasoning here you can always leave a comment. I will leave this pull request open so that I can revisit it if necessary.

**Comment by ajdevries** [comment]
Yeah, I agree. I think that you are right. So to fix the buffering, the method signature should be changed. Because you need the length of the bytes by forehand.

I used this library for another project and got things working with the changes. I need to backup databases using SCP, so for the larger files I'm not happy with the buffering.

**Comment by bramvdbogaerde** [comment]
Okay, I see the problem with the buffering.

So what I think that we can do in order to make this a better fit for large files:
* change the method signature of CopyFile, for API stability I will call this function "Copy"
* provide a special function which only accepts a os.File and passes its reader to "Copy", call this function "CopyFromFile"
* let CopyFile pass it's reader to "Copy" but let it calculate the size by using a buffer

What do you think?

**Comment by bramvdbogaerde** [comment]
I made some changes in the "dev" branch, would this cover your needs?

**Comment by ajdevries** [comment]
Yeah, great these changes look very good! Thanks for taking your time on this.

**Comment by bramvdbogaerde** [comment]
I will first test the dev branch and create a solution for your issue #3, and I will then merge in into master. Thanks for your feedback.

**Comment by bramvdbogaerde** [comment]
This functionality is already implemented in the "dev" branch.
Awaiting fix for issue #3 in order to merge dev with master.
